### PR TITLE
Smart sidebar

### PR DESCRIPTION
**Pull request by tajo**
Continues in https://github.com/cloudflare/doca-bootstrap-theme/pull/3 and adds
- interactive search
- automatic highlighting of the sections when you are scrolling

(you first need to merge https://github.com/cloudflare/doca-bootstrap-theme/pull/3 to see a proper diff of this PR)

screen shot 2017-02-01 at 10 02 43 pm

### CODE REVIEW

**Comment by handrews** [comment]
@tajo I approved and merged #3, want to rebase this for a better diff?

**Comment by tajo** [comment]
@handrews Rebased.

File: components/sidebar.js [hunk 1]
```diff
@@ -1,29 +1,130 @@
...
+          .map(link => link.get('html_id'))
+          .toJS()
+      );
+    }
+    , []);
+
+    const scrollTop = window.pageYOffset || document.documentElement.scrollTop ||
+                      document.body.scrollTop || 0;
+
+    // finds the first link that has top offset > top sroll position and breaks
```
Comment (handrews) [inline components/sidebar.js]: Typo: sroll vs scroll

File: components/sidebar.js [hunk 2]
```diff
@@ -1,29 +1,130 @@
...
+    let activeId = null;
+    for (let i = 0; i < ids.length; i++) {
+      if (offsetTop(document.getElementById(ids[i])) - 200 > scrollTop) {
```
Comment (handrews) [inline components/sidebar.js]: Where does 200 come from?

**Review by handrews — APPROVED** [review]
Generally looks good to me.  The 200 may be obvious to someone who has been doing UI work more recently than me, but I thought I'd ask.  Might be worth a comment?  The typo fix does not require a re-review :-)

**Comment by tajo** [comment]
Typo fixed.

> The 200 may be obvious to someone who has been doing UI work more recently than me

Good question. I wasn't sure for few minutes either. :)

The magic number 200 has been changed just to 30 and explained in the code comment:

```js
// a small offset so the coming section is highlighted a bit sooner
// before its main title touches the top of browser and starts disappearing
const VERTICAL_OFFSET = 30;
```

It was 200 because I copypasted it from doca-cf-theme and there is a top fixed menu that takes some vertical space as well.

**Comment by handrews** [comment]
@tajo looks good to me.

**Pull request merged**